src/app/components/homepage-components/navbar/input-field/input-field.component.html

```diff
@@ -24,6 +24,6 @@
   <div class="add-item">
     <img src="../../assets/user.png" alt="user-logo">
     <img src="../../assets/shopping_basket.png" alt="basket">
-    <p class="cart-items">{{productDetailService.addToCartItems}}</p>
+    <p class="cart-items">{{this.productDetailService.addToCartItems}}</p>
```

Comment:
do you need <this> before productDetailService ??

Reply:
Fixed using *ngIf directive. Now, when there are no added products, the basket is empty, so the number is not reflected.